src/tasks/set-connection-variables/set-connection-variables-v0/index.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as tl from 'azure-pipelines-task-lib/task';
+import { isRunningOnAgent } from "../../../params/auth/isRunningOnAgent";
+import {
+  EnvUrlVariableName,
+  ApplicationIdlVariableName,
+  ClientSecretVariableName,
+  TenantIdVariableName,
+  DataverseConnectionStringVariableName,
+  UserNameVariableName,
+  PasswordVariableName
+} from "../../../host/PipelineVariables";
+
+(async () => {
+  if (isRunningOnAgent()) {
+    await main();
+  }
+})().catch(error => {
+  tl.setResult(tl.TaskResult.Failed, error);
+});
+
+export async function main(): Promise<void> {
+  const authenticationType = tl.getInputRequired('authenticationType');
+  const endpointUrl = tl.getVariable(EnvUrlVariableName);
+
+  switch (authenticationType) {
+    case 'PowerPlatformSPN': {
+      const powerPlatformSPN = tl.getInputRequired(authenticationType);
+      const applicationId = tl.getEndpointAuthorizationParameterRequired(powerPlatformSPN, 'applicationId');
+      const clientSecret = tl.getEndpointAuthorizationParameterRequired(powerPlatformSPN, 'clientSecret');
+      const tenantId = tl.getEndpointAuthorizationParameterRequired(powerPlatformSPN, 'tenantId');
+
+      tl.setVariable(ApplicationIdlVariableName, applicationId, true);
+      tl.setVariable(ClientSecretVariableName, clientSecret, true);
+      tl.setVariable(TenantIdVariableName, tenantId, true);
+
+      const dataverseConnectionString = `AuthType=ClientSecret;url=${endpointUrl};ClientId=${applicationId};ClientSecret=${clientSecret}`;
+      tl.setVariable(DataverseConnectionStringVariableName, dataverseConnectionString);
```

Comment:
since the connStr contains the client secret, it must be flagged as secret GH variable as well